nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -17,14 +17,89 @@
 from nilearn._utils.exceptions import DimensionError
 from nilearn.image import get_data, new_img_like
 
+
 _TEST_DIM_ERROR_MSG = ("Input data has incompatible dimensionality: "
                        "Expected dimension is 3D and you provided "
                        "a 4D image")
+_TEST_SHAPE_ERROR_MSG = "Images have incompatible shapes."
+_TEST_AFFINE_ERROR_MSG = "Images have different affine matrices."
 
+INF = 1000 * np.finfo(np.float32).eps
 
-def test_generate_regions_ts():
-    """Minimal testing of generate_regions_ts()"""
 
+def test__check_shape_affine_label_img():
+    """Test to ensure correct check for valid shapes & affines of labels."""
+    shape = (8, 9, 10)
+    test_shape = (8, 9, 5)
+
+    # data
+    affine = np.eye(4)
+    test_affine = np.eye(4) * 2
+
+    # labels
+    labels_img = generate_labeled_regions(shape, 7, affine=affine)
+
+    # Ensure correct behaviour for valid data.
+    target_img = nibabel.Nifti1Image(np.zeros(shape), affine)
+    with pytest.warns(None):
+        signal_extraction._check_shape_affine_label_img(
+            labels_img, target_img)
+
+    # Smoke test to make sure an error is raised when shape is not correct.
+    target_img = nibabel.Nifti1Image(np.zeros(test_shape), affine)
+    with pytest.raises(ValueError, match=_TEST_SHAPE_ERROR_MSG):
+        signal_extraction._check_shape_affine_label_img(
+            labels_img, target_img)
+
+    # Smoke test to make sure an error is raised when affine is not correct.
+    target_img = nibabel.Nifti1Image(np.zeros(shape), test_affine)
+    with pytest.raises(ValueError, match=_TEST_AFFINE_ERROR_MSG):
+        signal_extraction._check_shape_affine_label_img(
+            labels_img, target_img)
+
+
+def test__check_shape_affine_maps_masks():
+    """Test to ensure correct check for valid shapes & affines of masks \
+    and maps."""
+    maps_shape = (2, 3, 4, 7)
+    mask_shape = (2, 3, 4)
+    test_mask_shape = (2, 3, 5)
+
+    affine = np.eye(4)
+    test_affine = 2 * affine
+
+    # Ensure correct behaviour for valid data without dim.
```

Comment:
The fact that this function (and the one above) are broken into several parts with comments suggests that you might want to split them in multiple test functions doing, if possible, a single assert each.

Reply:
split the tests for the extracted functions

the other tests should require the same treatment but I would rather do that after the reformatting